tests/defaultexecutor.cpp

```diff
@@ -180,6 +217,45 @@ TEST_F(DefaultExecutorTest, ThenWithException)
     executor->stop();
 }
 
+using DefaultExecutorDeathTest = DefaultExecutorTest;
+
+TEST_F(DefaultExecutorDeathTest, ObserveWithException)
+{
+    GTEST_FLAG_SET(death_test_style, "threadsafe");
+
+    auto executor = make_shared<DefaultExecutor>(milliseconds(10));
+    Default<Executor>::Setter execSetter(executor);
+
+    mutex mtx;
+    condition_variable cv;
+
+    // This will never become true, as the continuation will rethrow the exception
+    // thrown by the future passed to observe(). It only serves as barrier to block
+    // the execution until observe() throws.
+    bool success = false;
+
+    EXPECT_DEATH(
+        {
+            observe(getExceptionAsync<SomeKindOfError>(), [&](future<void> f) {
+                f.get();
+
+                // Because f.get() throws, the remaining code in this block
+                // is never executed.
```

Comment:
If we expect to never get to this code, why not add a `FAIL() << "Should not be executed";` line to incidate the issue to the person running the test?